Marlin/src/inc/Conditionals_post.h

```diff
@@ -1936,206 +1928,281 @@
   #endif
 #endif
 
-//
-// Endstops and bed probe
-//
-
-#define _HAS_STOP(A,N,M,C) ((A##_HOME_TO_##M || (C+0)) && PIN_EXISTS(A##N##_##M))
+/**
+ * Endstop and probe flags
+ * - Set USE_(AXIS)_(MIN|MAX) flags for each used endstop that has a pin, including those for DIAG0 state.
+ *   - Note: Dual X Carriage uses "X" and "X2" steppers, but X_MIN and X_MAX endstop states (i.e., not X2_MAX).
+ * - Set a HAS_(AXIS)_(MIN|MAX)_STATE flag for each endstop that has a state, including SPI Sensorless which don't use a pin.
+ * - Set a HAS_(AXIS)_STATE flag for each axis that has at least one state.
+ */
+#define _USE_STOP(A,N,M,C) ((A##_HOME_TO_##M || (C+0)) && PIN_EXISTS(A##N##_##M) && !A##_SPI_SENSORLESS)
 
-#if _HAS_STOP(X,,MIN,)
-  #define HAS_X_MIN 1
+#if _USE_STOP(X,,MIN,)
+  #define USE_X_MIN 1
+#endif
+#if _USE_STOP(X,,MAX,ENABLED(DUAL_X_CARRIAGE))
+  #define USE_X_MAX 1
+#endif
+#if USE_X_MIN || BOTH(X_SPI_SENSORLESS, X_HOME_TO_MIN)
+  #define HAS_X_MIN_STATE 1
 #endif
-#if _HAS_STOP(X,,MAX,ENABLED(DUAL_X_CARRIAGE))
-  #define HAS_X_MAX 1
+#if USE_X_MAX || BOTH(X_SPI_SENSORLESS, X_HOME_TO_MAX)
+  #define HAS_X_MAX_STATE 1
 #endif
-#if HAS_X_MIN || HAS_X_MAX
-  #define HAS_X_ENDSTOP 1
+#if HAS_X_MIN_STATE || HAS_X_MAX_STATE
+  #define HAS_X_STATE 1
 #endif
 
-#if _HAS_STOP(Y,,MIN,)
-  #define HAS_Y_MIN 1
-#elif _HAS_STOP(Y,,MAX,)
-  #define HAS_Y_MAX 1
+#if _USE_STOP(Y,,MIN,)
+  #define USE_Y_MIN 1
+#elif _USE_STOP(Y,,MAX,)
```

Comment:
USE_Y_MAX should be independent of USE_Y_MIN just like USE_X_MAX is.

Reply:
There is no such configuration that I know of. The X MIN and X MAX pin states are only ever used on the same machine when using `DUAL_X_CARRIAGE`.

Reply:
He he!!
Well, with me you will meet the first one. 
It may not be orthodox, but it is the configuration that I have and that works successfully for TMC2209. I have a single pin for #_MIN_PIN and #_MAX_PIN.
I had to adopt this solution due to the lack of free pins, and at the time it seemed the most logical thing to do since the detection of any of the limit switches would always come from the same pin on the controller (DIAG0). Why would I divide it if Was the software capable of detecting them independently?

Reply:
That's fine. So do you use the extra "endstop" for `SD_ABORT_ON_ENDSTOP_HIT`? Because otherwise leaving the endstops enabled doesn't do anything good, except cause the axis to get out of wack.

Part of the aim of this endstops refactor is to make it more explicit that there is an extra "endstop" but of course what you've got there is not actually an endstop because it can't be used for homing in the usual way. It's possible that a second endstop could be hacked together with a combination of `M120`, `G1`, `M400`, and `G92`, but that is not in the Marlin paradigm.

In fact, there has never been any formal support in Marlin for a two-endstop axis, so that is one of the things this PR may explore. For now I added `(AXIS)_SAFETY_STOP` as a way to retain two endstops on a single axis. However, it does not really deal with the problem or feature in a comprehensive way. In general, the endstops code in Marlin has not been given a proper design makeover.